pkg/pillar/cmd/baseosmgr/handleverifier.go

```diff
@@ -49,15 +50,85 @@ func lookupVerifierConfig(ctx *baseOsMgrContext, objType string,
 	return &config
 }
 
+// whether the object contains certs reference
+func containsCerts(safename string, ss *types.StorageConfig) bool {
+	cidx := 0
+	// count the number of cerificates in this object
+	if ss.SignatureKey != "" {
+		cidx++
+	}
+	for _, certURL := range ss.CertificateChain {
+		if certURL != "" {
+			cidx++
+		}
+	}
+	// if no cerificates, return
+	if cidx == 0 {
+		log.Infof("containsCerts() for %s, no certificates configured\n", safename)
+		return false
+	}
+	return true
+}
+
+// check the cert object status
+func checkCertsStatusForObject(ctx *baseOsMgrContext, uuidStr string,
+	safename string, ss *types.StorageStatus) error {
+
+	certObjStatus := lookupCertObjStatus(ctx, uuidStr)
+	// certificates are still not ready, for processing
+	if certObjStatus == nil {
+		return errors.New("certObj: Not ready")
+	}
+	if ss.SignatureKey != "" {
+		for _, certObj := range certObjStatus.StorageStatusList {
+			if certObj.Name == ss.SignatureKey {
+				if certObj.Error != "" {
+					ss.Error = certObj.Error
+					ss.ErrorTime = certObj.ErrorTime
+					ss.ErrorSource = pubsub.TypeToName(types.VerifyImageStatus{})
+					return errors.New("certObj: Error")
+				}
+				if certObj.State != types.DELIVERED {
+					return errors.New("certObj: Not ready")
+				}
+			}
+		}
+	}
+
+	for _, certURL := range ss.CertificateChain {
+		if certURL != "" {
+			for _, certObj := range certObjStatus.StorageStatusList {
+				if certObj.Name == certURL {
+					if certObj.Error != "" {
+						ss.Error = certObj.Error
+						ss.ErrorTime = certObj.ErrorTime
+						ss.ErrorSource = pubsub.TypeToName(types.VerifyImageStatus{})
+						return errors.New("certObj: Error")
+					}
+					if certObj.State != types.DELIVERED {
+						return errors.New("certObj: Not ready")
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
 // If checkCerts is set this can return an error. Otherwise not.
-func createVerifierConfig(ctx *baseOsMgrContext, objType string, safename string,
-	sc *types.StorageConfig, checkCerts bool) error {
+func createVerifierConfig(ctx *baseOsMgrContext, uuidStr string, objType string,
```

Comment:
This seems to be repeated... Why not have some utility routines and invoke from both places?

Reply:
moved to common method.